src/foam/core/ContextAgentRunnable.java

```diff
@@ -32,6 +37,14 @@ public void run() {
         Thread.currentThread().setName(description_);
       }
       agent_.execute(x_);
+    } catch (RuntimeException e) {
+      Logger logger = (Logger) x_.get("logger");
```

Comment:
Replace all of this with an EventRecord.  The eventrecord will create an log, and alarm and notification if warning or error.